partition_table.go

```diff
@@ -521,7 +523,7 @@ func (p *PartitionTable) loadMessages(ctx context.Context, cons sarama.Partition
 			}
 
 			lastMessage = time.Now()
-			if err := p.storeEvent(string(msg.Key), msg.Value, msg.Offset, msg.Headers...); err != nil {
+			if err := p.storeEvent(string(msg.Key), msg.Value, msg.Offset, headers.FromSarama(msg.Headers)); err != nil {
```

Comment:
this will be called on every message being loaded, but actually we wanted to avoid converting all headers, right? This makes the lazyloading implemented in the UpdateContext kind of useless, as the field `saramaHeaders` is never actually instantiated, or am I missing something?

Reply:
_sigh_

Thank you for stopping me from merging suboptimal code. I should have been more thorough with this PR. I'll fix those asap.